vxyowsup/tests/test_whatsapp.py

```diff
@@ -120,6 +136,17 @@ def receive(self, data):
         '''
         self.toUpper(data)
 
+    def send_ack(self, node):
+        # TODO: rather send IncomingAckProtocolEntity and test extra fields
+        ack = AckProtocolEntity(_id=node['id'], _class='message')
+        self.receive(ack.toProtocolTreeNode())
+
+    def send_receipt(self, node):
+        # msg_class=None defualt indicates "delivered"
+        # alt: msg_class='read'
+        receipt = IncomingReceiptProtocolEntity(_id=node['id'], _from=node['to'], timestamp=node['id'].split('-')[0], type='read')
```

Comment:
Well, it's equivalent to the original timestamp of the message. That's how the ids are created. Or should the timestamp be for the receipt? I got the impression the timestamp of the message was passed down to the receipt but I am no longer sure.